concepts/comparisons/about.md

```diff
@@ -1,199 +1,291 @@
+# hello < goodbye
+>>> '你好' < '再见'  
 True
->>> # check ord() of first letters
+
+# ord() of first characters
 >>> ord('你'), ord('再')
 (20320, 20877)
->>>
->>>
->>> # let's try Korean words
->>> '예쁜' < '아름다운'  # pretty < beautiful
+
+# ord() of second characters
+>>> ord('好'), ord('见')
+(22909, 35265)
+
+# And with Korean words.
+# Pretty < beautiful.
+>>> '예쁜' < '아름다운'  
 False
+
 >>> ord('예'), ord('아')
 (50696, 50500)
 ```
 
-Collections like `list`, `set`, `tuple` and `dict` can also be compared -- provided they are of the same `type`, have the same length, and each _**pair**_ of corresponding elements within the collection are comparable.
+
+## Comparing Container Data Types
+
+Container data types (_`lists`, `tuples`, `sets`, `dicts`, etc._) also compare [_lexicographically_][lexographic order] - they are equal if both containers have the same data **and** the same data types (_in the case of `lists` and `tuples`, they must also have the same **ordering**_), unequal otherwise.
+
 
 ```python
+# Comparing lists
 >>> [1, 2] == [1, 2]
 True
->>> (1, 2) == [1, 2]
+
+# But if the data is not in the same order, they are not equal.
+>>> [2, 1] == [1, 2]
 False
+
+# The same holds true for tuples
+>>> (3,4,5) == (5,4,3)
+False
+
+# Length is also compared
 >>> [1, 2] < [1, 2, 3]
 True
->>> # comparison of dicts
+
+# Comparing dicts
 >>> {'name': 'John', 'age': 19} == {'name': 'John', 'age': 18}
 False
+
 >>> {'name': 'John', 'age': 19} == {'name': 'John', 'age': 19}
 True
 ```
 
+
+## Comparison Chaining
+
+Comparison operators can be chained _arbitrarily_.
+Note that the evaluation of an expression takes place from `left` to `right`.
+For example, `x < y <= z` is equivalent to `x < y` `and` `y <= z`, except that `y` is evaluated **only once**.
+In both cases, `z` is _not_ evaluated **at all** when `x < y` is found to be `False`.
+This is often called `short-circuit evaluation` - the evaluation stops if the truth value of the expression has already been determined.
+
+`Short circuiting` is supported by various boolean operators, functions, and also by comparison chaining in Python.
+Unlike many other programming languages, including `C`, `C++`, `C#`, and `Java`, chained expressions like `a < b < c` in Python have a conventional [mathematical interpretation][three way boolean comparison] and precedence.
+
+
+```python
+>>> x = 2
+>>> y = 5
+>>> z = 10
+
+>>> x < y < z
+True
+
+>>> x < y > z
+False
+
+>>> x > y < z
+False
+```
+
+
 ## Identity comparisons
 
-The operators `is` and `is not` test for an object's _identity_.
-An object's identity is determined using the `id()` function.
+The operators `is` and `is not` test for object [_identity_][object identity], as opposed to object _value_.
+An object's identity never changes after creation and can be found by using the [`id()`][id function] function.
+
+\<apple\> `is` \<orange\> evaluates to `True` if _**and only if**_ `id(<apple>)` == `id(<orange>)`.
+apple `is not` orange yields the inverse.

```

Comment:
It looks like the code delimiter changed here as well, perhaps unintended.

Reply:
Same as above -- not sure what reads better, but a whole block doesn't read well to me.